tests/test_basic_ops_dict.py

```diff
@@ -72,12 +72,27 @@ def test_mandatory_value() -> None:
         c.a
 
 
+def test_mandatory_value_int_key() -> None:
```

Comment:
I think this test (as well as other tests added in this commit) would better be created by parameterizing the existing test operating on strings (here: `test_mandatory_value()`). This way, if we want to support float / bool keys as well for instance, it will be much more straightforward than to copy again the same set of tests.

Reply:
Done in commit https://github.com/omry/omegaconf/pull/454/commits/35c796a68ec4e8580fa2914ca2ca59f077184941

Reply:
Nice, thanks! I didn't expect all those `if isinstance(key, str)` cases (since `cfg[key]` works for strings too) but it doesn't hurt to test them I guess :) You could have also used a helper function to switch between the two ways to access keys without having to repeat these checks everywhere -- though your tests actually cover more access cases this way, so not asking you to do that.